neurodsp/aperiodic/dfa.py

```diff
@@ -0,0 +1,145 @@
+    The relationship of this fluctuation measure across window sizes provides
+    information on the fractal properties of a signal.
+
+    Measures available are:
+    - DFA: detrended fluctuation analysis
+        - computes ordinary least squares fits across signal windows
+    - RS: rescaled range
+        - computes the range of signal windows, divided by the standard deviation
+
+    Empirically, DFA seems to work for a larger range of exponent values.
+    """
+
+    # Get log10 equi-spaced scales and translate that into window lengths
+    t_scales = np.logspace(np.log10(min_scale), np.log10(max_scale), n_scales)
+    win_lens = np.round(t_scales * fs).astype('int')
+
+    # Calculate culmulative sum of the signal
+    sig_walk = sp.cumsum(sig - np.mean(sig))
+
+    # Step through each scale and get RMS
+    fluctuations = np.zeros_like(t_scales)
+    for idx, win_len in enumerate(win_lens):
+        if method is 'dfa':
+            fluctuations[idx] = compute_detrended_fluctuation(sig_walk, win_len=win_len, deg=deg)
+        elif method is 'rs':
+            fluctuations[idx] = compute_rescaled_range(sig_walk, win_len=win_len)
+        else:
+            raise ValueError('Fluctuation method not understood.')
+
+    # Calculate the relationship between between fluctuations & time scales
+    exp = np.polyfit(np.log10(t_scales), np.log10(fluctuations), deg=1)[0]
+
+    return t_scales, fluctuations, exp
+
+
+def compute_rescaled_range(sig, win_len):
+    """Compute rescaled range of a given time-series at a given scale.
+
+    Parameters
+    ----------
+    sig : 1d array
+        Time series.
+    win_len : int
+        Window length for each rescaled range computation.
+
+    Returns
+    -------
+    rs : float
+        Average rescaled range over windows.
+    """
+
+    # Gather windows & vectorize, so we can call math functions in one go
+    n_win = int(np.floor(len(sig) / win_len))
+    sig_rect = np.reshape(sig[:n_win * win_len], (n_win, win_len)).T
+
+    # get back the sig by taking the derivative of sig_walk
+    X = np.concatenate((sig[:1], np.diff(sig)))
+    X_rect = np.reshape(X[:n_win * win_len], (n_win, win_len)).T
+
+    # Calculate rescaled range as range divided by std, and take mean across windows
+    rs_win = np.ptp(sig_rect, axis=0) / np.std(X_rect, axis=0)
+    rs = np.mean(rs_win)
+
+    return rs
+
+
+def compute_detrended_fluctuation(sig, win_len, deg=1):
+    """Compute detrended fluctuation of a time-series at the given window length.
+
+    Parameters
+    ----------
+    sig : 1d array
+        Time series.
+    win_len : int
+        Window length for each detrended fluctuation fit.
+    deg : int, optional, default=1
+        Polynomial degree for detrending.
+
+    Returns
+    -------
+    det_fluc : float
+        Measured detrended fluctuaiton, as the average error fits of the window.
+    """
+
+    # Gather windows & vectorize, so we can call math functions in one go
```

Comment:
This is actually the same procedure that is now a helper function in `neurodsp.utils.data`, called `split_signal`, so we can use that, which gets rid of some duplicated code.

Reply:
great!

Reply:
Updated to use `split_signal`.